### DISCUSSION TITLE
Trigger sync from command line (e.g. Termux)

### DISCUSSION THREAD

**Discussion by doak** [orgzly-revived/orgzly-android-revived#144] [Q&A] [answered]
Hi all,

what is the best way to trigger a sync from command line (i.e. _Termux_)?

TL;DR:
My understanding is that current implementation (mainly the manifest) does not allow a sync to be triggered form _Termux_. But I don't see why this could worked a few week ago, though.

Background:
I had been using _Termux_ to watch for file changes and trigger a sync of _Orgzly_ automatically. This way it was possible to provide some kind of automatic sync to local storage (which itself had been synced with _Syncthing_ across several devices).
I switched to _Orgzly Revived_ just a few weeks ago. It took some time to adapt the `am` command but it worked.
Now my Android had been updated to 14, and it seems that it fails since then.

Details:
Initially, that is for _Orgzly_, I have used this command: ([Source](https://github.com/orgzly/orgzly-android/issues/999) for this command.)
`am broadcast -a com.orgzly.intent.action.SYNC_START com.orgzly/.android.ActionReceiver`
I adapted that for _Orgzly Revived_ like that:
`am broadcast -a com.orgzly.intent.action.SYNC_START com.orgzlyrevived/com.orgzly.android.ActionReceiver`

But is does not work anymore. Return code is 255.
I've also tried some other ways, but I don't have a clue about these Android specifics unfortunately, There is also a `com.orgzly.android.ui.SyncShortcutActivity` which looks promising, but I was not able to trigger that either. I also played around with `am start`.

Btw., _Termux_ has the permission to "draw above other apps" in case this is relevant.

I played around with _Activity Launcher_ (`de.szalkkoswki.activitylauncher`). That application allows to launch activities with one click and is able to create shortcuts for these. Interesting enough, there is a small lock symbol next to `com.orgzly.android.ui.SyncShortcutActivity` and an error toast appears (something about denied permissions) if launched directly. The same happens even when launching from my home screen using a created shortcut, though the "official" shortcut to trigger sync works.
It seems the small lock symbol represents "non-exported" activities. ~I vague remember that I read about a change in this context for Android 14.~ (I've found no evidence on [official documentation](https://developer.android.com/about/versions/14/behavior-changes-all).) But how does my launcher (_Kiss_) launch this? It is a "home app" – perhaps these are allowed to even start non-exported activities. The [documentation about exported activities](https://developer.android.com/privacy-and-security/risks/android-exported) tells that "privileged system components" are allowed. It makes imho sense that a launcher is one of them.
But this does not explain why it has worked before the update to Android 14. I've found no evidence that the default value of this attribute had changed lately.

Any help is appreciated.
Thanks.

**Comment by doak** [comment, marked as answer]
> `am broadcast -a com.orgzly.intent.action.SYNC_START com.orgzlyrevived/com.orgzly.android.ActionReceiver`

Out of nothing this works again. Most likely I either messed it up or something on my mobile phone was behaving strangely.

**Comment by danielkrajnik** [comment]
Thanks for sharing this, I've had no idea about `am` command - looks really useful and I will definitely give it a try myself. Good luck with your use case.

**Comment by doak** [comment]
I am wondering why I haven't referenced the script as well, which uses this command to keep Orgzly data in sync with filesystem (without polling using `inotifywait`): [watch-for-orgzly](https://gitlab.com/doak/orgzly-watcher/-/blob/1a7f8daed51775134995f639fc5cecf7127b7461/watch-for-orgzly#L128)

**Comment by dvictori** [comment]
I've been trying to sync Orgzly in the background with my Nextcloud webdav. After reading this post and looking at some other online text, I saw that some have achieved this with `tasker`. But I use `Automate`. So, after messing around, I was able to get it working using Automate. Though I would share, just in case.

In automate we use the "Broadcast Send"
![image](https://github.com/user-attachments/assets/ac117336-6d29-4860-883d-f705236f7ccd)
